README.md

```diff
@@ -1,2 +1,168 @@
+|triggerDate|optional DateTime|-|A date to trigger a notification on the device. If not set, then there is no notification to trigger. The notification should be triggered by the server. TBD: Are there cases where a message would trigger multiple times or event- rather than date-driven?|
+|completionDate|optional DateTime|-|Specifies when a message has been completed. TBD: Messages containing a completionDate may either be hidden on user's devices or be shown crossed out.|
+|title|LocalizedText|e.g. "Watch Welcome Video in Education Page."|May be localized.|
+|description|optional LocalizedText|e.g. "The video shows how you will be able to use this app."|May be localized.|
+|action|optional string|e.g. "engage-hf:/videoSections/engage-hf/videos/welcome"|Format to be defined, needs to be decoded on a device and performed on tap of the message.|
+
+### /users/$userId$/allergyIntolerances/$allergyIntoleranceId$
+
+Based on [FHIR AllergyIntolerance](https://hl7.org/fhir/R4B/allergyintolerance.html), the following properties may be used, while additional properties are ignored by the Engage-HF system.
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|id|string|-|[Resource](https://hl7.org/fhir/R4B/resource.html): Logical id of this artifact|
+|type|optional [allergyIntoleranceType](https://hl7.org/fhir/R4B/valueset-allergy-intolerance-type.html)|e.g. "allergy", "intolerance"||
+|code|CodableContent|e.g. "{"coding":[{"system":"https://hl7.org/fhir/R4B/valueset-allergyintolerance-code.html","code":"293963004","display":"Cardioselective beta-blocker allergy"}],"text":"Cardioselective beta-blocker allergy"}"|Uses either [AllergyIntoleranceCode](https://hl7.org/fhir/R4B/valueset-allergyintolerance-code.html), `medicationId` as used in /medications/$medicationId$ and/or `medicationId` as used in /medicationClasses/$medicationClassId$.|
+|patient|string|e.g. "allergy", "intolerance"|`userId` as used in /users/$userId$ and related collections.|
+
+### /users/$userId$/medicationRequests/$medicationRequestId$
+
+Based on [FHIR MedicationRequest](https://hl7.org/fhir/R4B/medicationrequest.html), the following properties may be used, while additional properties are ignored by the Engage-HF system.
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|id|string|-|[Resource](https://hl7.org/fhir/R4B/resource.html): Logical id of this artifact|
+|type|optional [allergyIntoleranceType](https://hl7.org/fhir/R4B/valueset-allergy-intolerance-type.html)|e.g. "allergy", "intolerance"||
+|code|CodableContent|e.g. "{"coding":[{"system":"https://hl7.org/fhir/R4B/valueset-allergyintolerance-code.html","code":"293963004","display":"Cardioselective beta-blocker allergy"}],"text":"Cardioselective beta-blocker allergy"}"|Uses either [AllergyIntoleranceCode](https://hl7.org/fhir/R4B/valueset-allergyintolerance-code.html), `medicationId` as used in /medications/$medicationId$ and/or `medicationId` as used in /medicationClasses/$medicationClassId$.|
```

Comment:
I think allergy intolerance code might not be needed here.

Reply:
It depends. There are two ways we can go about this, I think. In general, we just need to be consistent with the ones that may be used in the web dashboard and the ones that are checked for in the algorithms.

Option 1: We pretty much exclusively use AllergyIntoleranceCode and easily interface with an EHR system in the future. I dug into this a little bit and it might be quite complex though to decide with existing identifiers would be required for each algorithm. Since these ids also seem to have a hierarchy that isn't necessarily represented in the id itself (e.g. a certain prefix), we would probably need to check for all kinds of ids. For the first development step, we could however limit our development efforts to the ones that we allow for input on the web dashboard. 

Option 2: We use our own ids, which cannot easily interface with an EHR system. This would simplify the implementation of the system initially, but might require some form of overhaul, if we were to ever interface with an EHR system.

Reply:
I would lean to Option 1 with a clear limitation to what we can enter in the web frontend. What do you think @vishnuravi?

Reply:
I added the list of codes to the allergyIntolerances section. It is not entirely complete, but for the remaining few codes, there are TBD comments for now